### PR TITLE
Remove deprecated BaseTracer#extractWithGlobalPropagators() methods

### PR DESCRIPTION
**Pull request by mateuszrzeszutek**

### CODE REVIEW

File: instrumentation/rmi/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/rmi/context/ContextPayload.java [hunk 1]
```diff
@@ -42,9 +34,8 @@ public ContextPayload(Map<String, String> context) {
     this.context = context;
   }
 
-  public static ContextPayload from(Span span) {
+  public static ContextPayload from(Context context) {
     ContextPayload payload = new ContextPayload();
-    Context context = Context.current().with(span);
     GlobalOpenTelemetry.getPropagators().getTextMapPropagator().inject(context, payload, SETTER);
```
Comment (jkwatson) [inline instrumentation/rmi/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/rmi/context/ContextPayload.java:39]: there's still a global call in here...that seems like a problem.

Comment (mateuszrzeszutek) [reply instrumentation/rmi/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/rmi/context/ContextPayload.java:39]: Yep - currently all `inject()` calls are using global propagators. That's going to be the subject for my next PR 😄

File: instrumentation/rmi/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/rmi/context/ContextPayload.java [hunk 2]
```diff
@@ -24,12 +22,6 @@
 
   private static final Logger log = LoggerFactory.getLogger(ContextPayload.class);
 
-  private static final Tracer TRACER = GlobalOpenTelemetry.getTracer("io.opentelemetry.auto.rmi");
```
Comment (jkwatson) [inline instrumentation/rmi/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/rmi/context/ContextPayload.java:27]: this is a good change, but I don't see how it's related to the topic of the PR.

Comment (mateuszrzeszutek) [reply instrumentation/rmi/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/rmi/context/ContextPayload.java:27]: It was just a bit of unused code - I thought that removing it in a separate PR would be a bit of excess of form, but you're absolutely right that it't not related to this PR.

**Review by jkwatson — APPROVED** [review]

**Review by trask — APPROVED** [review]

**Review by anuraaga — APPROVED** [review]

**Review by iNikem — APPROVED** [review]

**Pull request merged**